### ISSUE TITLE
Inconsistent/ill-defined recursion behavior

### ISSUE THREAD

**Issue by Arithmomaniac** [tsdservices/Gravity#21]
I was refactoring the RsapiDao in preparation for working on #8, and notices something strange.
You can see the refactor [here](https://github.com/Arithmomaniac/Gravity/commit/7e28c7bce94cf51073c11407c797e8fb95a10931). 

Most of the recursion can be refactored out to the following method:

```csharp
private T GetHydratedDTO<T>(RDO objectRdo, ObjectFieldsDepthLevel depthLevel) 
    where T : BaseDto, new()
{
    T dto = objectRdo.ToHydratedDto<T>();

    switch (depthLevel)
    {
        case ObjectFieldsDepthLevel.FullyRecursive:
            PopulateChildrenRecursively<T>(dto, objectRdo, depthLevel);
            return dto;
        default:
            return dto;
    }
}
```

But one method seems to use different logic:

```csharp
public T GetRelativityObject<T>(int artifactId, ObjectFieldsDepthLevel depthLevel)
    where T : BaseDto, new()
{
    var objectRdo = GetRdo(artifactId);
    var dto = objectRdo.ToHydratedDto<T>();

    switch (depthLevel)
    {
        case ObjectFieldsDepthLevel.OnlyParentObject:
            return dto;
        default:
            PopulateChildrenRecursively<T>(dto, objectRdo, depthLevel);
            return dto;
    }
}
```

In addition to being inconsistent, it seems at first glance that neither rule covers what one would expect for `ObjectDepthsFieldLevel.FirstLevelOnly` - mainly, that there would be one level of recursion, but that level would hydrate the DTOs with `ObjectFieldsDepthLevel.OnlyParentObject`.

We should clarify what, exactly, we expect each member of the enum to do, and fix/refactor the recursion calls accordingly.

**Comment by ggachevski** [comment]
Yep, this guy needs some refactoring. It is more of a necessity than an actual bug, but work is needed. I am marking as "help needed". Shooting over to @Arithmomaniac, as he is very willing.

**Comment by Arithmomaniac** [comment]
While we're refactoring,why do some methods have the signature `DoSomethingDTO(s)` and others `DoSomethingRelativityObject(s)`? Can the signatures be standardized?

**Comment by ggachevski** [comment]
Yes, we’ve been neglecting this refactoring for some time. Ideally, we fancy the “RelativityObject” naming more than the “DTO”.

We are glad you feel the same way. Go wild!

**Comment by Arithmomaniac** [comment]
fea

**Comment by Arithmomaniac** [comment]
I deferred the renaming of the classes, because I don't want to start breaking the interface until I have a way to test Gravity set up.

**Comment by jacobmalliet** [comment]
Fixed in #26
